### DISCUSSION TITLE
Typing the starting characters of an item in a selected list should jump you to it

### DISCUSSION THREAD

**Discussion by grantfitzsimmons** [specify/specify7#5572] [Feature Requests]
**Is your feature request related to a problem? Please describe.**
When navigating a long list of table names or items it can be frustrating to find the one you're looking for. In Specify 6 typing the first few characters of an item in a list jumps you to it.

This is especially helpful when browsing an extremely long list of items (schema config, queries, record sets).

**Describe the solution you'd like**
I would like to either search/filter by my input or begin typing characters of a given field and automatically select it.

This behavior could be similar to how a file browser handles finding a file when you select anything and begin typing characters.

In Specify 6 you can open the Schema Config and type the name of the table you wish to find and it will jump you there.

**Describe alternatives you've considered**
Textbox that allows me to filter the contents of a dialog (display items that contain some part of my input)

**Comment by grantfitzsimmons** [comment]
![Specify4GB_2zccgfPvlG](https://user-images.githubusercontent.com/37256050/163621936-2a02159f-5dad-4006-8736-f9a780ef6379.gif)

Difficult to demonstrate, but this is what happens when I type while in the **Tables** list.

**Comment by maxpatiiuk** [comment]
Replacing `<ul>` with `<select>` where possible would fix this - note this won't support displaying table icons.

If need to also display table icons, may have to integrate with something like https://headlessui.com/react/listbox

Otherwise, have to wait for `<selectmenu>` to be supported on the web - https://open-ui.org/prototypes/selectmenu/
